### ISSUE TITLE
Can't patch resources when in failed state

### ISSUE THREAD

**Issue by lewismarshall** [UKHomeOffice/kb8or#69]
May have to patch in MD5 ONLY when resource deploy successful?

Kubectl can't patch failed replication controllers?

```
Error (exit code:'1') running 'kubectl patch ReplicationController jenkins -p '{"metadata":{"labels":{"kb8_md5":""}}}'':
Error from server: an error on the server has prevented the request from succeeding (patch replicationcontrollers jenkins)
```

**Comment by lewismarshall** [comment]
@allandegnan This issue prevents the failing pods being redeployed (as no change is detected in subsequent deploys).

It tries to remove the MD5 AFTER the pods is detected as failed. Not sure how to fix this as not sure why I can't patch - yet...

**Comment by lewismarshall** [comment]
I think I'll update this - should be able to use `kubectl label` instead of patch...

**Comment by lewismarshall** [comment]
kubectl label doesn't work (same error as above). Haven't looked into why.

Will have to test health of RC's that are already deployed and redeploy if required.

**Comment by lewismarshall** [comment]
Plan:
1. On failure, mark pods, NOT controllers (with label) as dirty
2. On deploy update is_dirty? method to check label above.

**Comment by lewismarshall** [comment]
Fixed by #74.
